File: MyApp/Models/Ballot/IBallot.cs
using System.Collections.Generic;

namespace MyApp.Models.Ballot
{
    public interface IBallot
    {
        int Id { get; set; }
        string Title { get; set; }
        string Description { get; set; }
        List<Candidate> Candidates { get; set; }
        BallotType Type { get; }
        bool ValidateVote(object vote);
    }

    public enum BallotType
    {
        YesNo,
        RankedChoice,
        MultipleChoice
    }
} 

File: client/src/services/ballot/IBallot.ts
import { Candidate } from '../candidateService';

export interface IBallot {
    id: number;
    title: string;
    description: string;
    candidates: Candidate[];
    type: BallotType;
    render(): JSX.Element;
    validate(vote: any): boolean;
}

export enum BallotType {
    YesNo = 'YesNo',
    RankedChoice = 'RankedChoice',
    MultipleChoice = 'MultipleChoice'
} 